app.js

```diff
@@ -13,22 +13,31 @@ var config = {
 
 require('./index.js').connect(function (context) {
 
+    context.fs.readdir(context.path.join(__dirname, 'api', 'models'), (err, files) => {
+        //Load ALL modules.
+        //Don't get confused by weird logging order! console.log is async and will mix up the actual loading order.
+        files.forEach(file => {
+            context.component('models').module(file.replace('.js',''));
```

Comment:
It's being cached within the context. This does get completely reset once we restart the server since the context will be newly created, too.
Let's look at this piece of code in [index.js](https://github.com/MusicConnectionMachine/api/blob/2f2c0e478c9843930366756351e5da44b8947cb4/index.js#L24-L41):
```
        context.component = function(componentName) {
            if (!context[componentName]) {
                context[componentName] = {};
            }

            return {
                module: function(moduleName) {
                    if (!context[componentName][moduleName]) {
                        console.log('Loading component ' + componentName);
                        context[componentName][moduleName] = require(path.join(__dirname, "api", componentName, moduleName))(context,
                            componentName, moduleName);
                        console.log('LOADED ' + componentName + '.' + moduleName);
                    }

                    return context[componentName][moduleName];
                }
            }
        };
```
When we call e.g. context.component('models').module('websites'). The first time we call that, the function will load the website module and store it as an attribute in the context. The second time we call that function context will not reload the module but return the module that was already stored in context before.